### ISSUE TITLE
how to have each vhost has its own acces log and error log?

### ISSUE THREAD

**Issue by UlasSAYGINIM** [h2o/h2o#2228]
I couldnt understand it. access log is only global directive and there is no any other directive that i can see to set log files for vhost individually.

wiki page has information but it is confusing when i looked the default config file on freebsd installation.
for example , access log information is written on top of the config file but in wiki page, it is 
bottom of host configs like this one. i also added this directive end of the host directive for only access log and error log but i couldnt see any error for config and h2o web server does not use it these too. it writes log on default location /var/logs/h2o/

how can i handle logs more effectively, for example to record logs into database like postgresql.

 https://github.com/h2o/h2o/wiki/redirect-HTTP-to-HTTPS

**Comment by UlasSAYGINIM** [comment]
/usr/local/etc/h2o/conf.d/bsdn1com.conf
  "www.bsdn1.com":
    listen:
      port: 80
    listen:
      port: 443
      ssl:
        certificate-file: /usr/local/etc/ssl/bsdn1com.pem
        key-file:         /usr/local/etc/ssl/key.pem
    paths:
      "/":
        file.dir: /usr/local/www/bsdn1com/web
access-log: /usr/local/www/bsdn1com/logs/access-log
error-log: /usr/local/www/bsdn1com/logs/error-log

**Comment by utrenkner** [comment]
While [``error-log``](https://h2o.examp1e.net/configure/base_directives.html#error-log) is a global-only configuration option, [``access-log``](https://h2o.examp1e.net/configure/access_log_directives.html) can be applied on these levels: global, host, path, extension. 

Thus, there is nothing that prevents you from defining an access log for each host separately.

**Comment by UlasSAYGINIM** [comment]
why the error log is globally? i dont get it. this log feature differences.
how the control panels will get the logs for each vhost individually? and seperation is also makes sense for website level. i couldnt understand its function. can you explain?

in addition this , the link below is not working for seperated log files but seems confusing, am i wrong?
https://github.com/h2o/h2o/wiki/redirect-HTTP-to-HTTPS
thank you.

**Comment by proyb6** [comment]
What is the *control panels* are you referring to?

**Comment by UlasSAYGINIM** [comment]
mine i am going to implement :) if i can achieve to work with h2o , i would like to use h2o as primary web server. i think it worth to do. i spoke , in a manner of speaking, someone can do if they need.
thank you for question @proyb6.

**Comment by proyb6** [comment]
@UlasSAYGINIM Interesting project, may I suggest it may also worth a try Golang into your control panel if the web applications can benefit from zero allocation logging performance with Uber zap or zerolog.

**Comment by UlasSAYGIN** [comment]
i am sorry about late response @proyb6 , of course you can suggest anything you want it. 
i didnt work with golang , therefore i dont know this point you mention about it. you mean i may need to write control panel in golang or provide golang features for better service for web applications? can you tell about it more in detail? thank you.

**Comment by UlasSAYGINIM** [comment]
yes sure you can suggest anything you like. but can you explain more about what features you want?
I didnt use golang and i couldnt understand your wish about control panel feature.

**Comment by UlasSAYGINIM** [comment]
@proyb6  can you say which zerolog? and it seems good idea thanks a lot.
